RELEASING.md

```diff
@@ -0,0 +1,78 @@
+# OpenSearch Querqy Plugin Releases
+
+This document contains details about the release strategy for OpenSearch Querqy Plugin. 
+We discuss things like branching and release cycles for easy maintenance in the long term.
+
+OpenSearch Querqy Plugin coupled with two opensource projects: Querqy and OpenSearch. 
+Both of these projects have different release cycles, versioning and package deployment strategies. 
+This document chooses the best of both worlds and defines a way to ensure the community plugins have a smoother blend to the OpenSearch release strategy.
+
+## 1. Branching
+
+### 1.1 Branch names
+* Querqy OpenSearch plugin branch names are as follows: 
+
+  * `main`: contains the current release in progress, currently 2.3.0. This is the branch where all merges take place, and code moves fast.
+  * `2.2`: contains the previous released code, i.e. 2.2.0. No further feature updates are done to this branch. 
+  * `2.3`: The current release, currently 2.3.0. In between minor releases, only hotfixes (e.g. security) are backported to 2.3. 
+    The next release out of this branch will be 2.3.1.
+    
+* __Note__: OpenSearch follows [semver](https://semver.org/) for versioning the releases. Querqy OpenSearch plugin follows the same.
+
+
+### 1.2 Release Tags
+
+* The release tag names are a combination of Querqy-lucene version and the suported OpenSearch version. 
+  For example, plugin built for `OpenSearch 2.3.0` with `Querqy-Lucene 5.4.lucene900.1`, will have the tag name: `querqy-opensearch-5.4.l900.1.os2.3.0`
+* Keeping tag names this way, help us to see the plugin's compatibility with OpenSearch and Querqy-Lucene project.
+* Release tags contain, assets for source-codes and pre-built zip artifacts. 
+* Hotfixes and security patches will have a patch suffix at the end of the release tag example: `querqy-opensearch-5.4.l900.1.os2.3.0-patch1`
+
```

Comment:
While this naming scheme reflects our dependencies very well, it would not allow us to version `querqy-opensource` development itself, eg. if we changed the query DSL; we might also decide to "pass through" features from `querqy-lucene` in a later version and not right when they become available (for example, the latest `querqy-lucene` version allows to chose between additive and multiplicative boosts for UP/DOWN rules and other rewriter output, which we didn't include in the version for os 2.0).

I would thus suggest: `querqy-opensearch-<major.minor>.os2.3.0-patch1` (or even `querqy-opensearch-<major.minor>.os230.1` - which would be equivalent to how we handle it for ES. So we'd start with `querqy-opensearch-1.0.os2.3.0` (or `querqy-opensearch-1.0.os230.0`). This would also allow us to create a `querqy-opensearch-1.0.os200.0` for users who haven't updated their OS version yet.

Reply:
Thanks for the comments 😄 
My intention behind keeping querqy-lucene version in the tag name was to not maintain another version matrix. But, as you specifically mentioned this is inevitable. I like this suggestion: `querqy-opensearch-<major.minor>.os2.3.0-patch1` I allows us to keep OS version and plugin version separate. Will update the doc with this change.

I'm particularly away from the usual ES convention of merging `semver` numbers in `querqy-opensearch-1.0.os230.0` This may be problematic in future if we have 2 digit version numbers. os2130 -> can mean OS 2.13.0 or OS 21.3.0.

Reply:
I might subconsciously have blanked out the problem and hoped to be retired by the time when `OS 21.3.0` gets released ;-) - let's keep the dots in!

Reply:
updated the doc!